src/cryptsetup/cryptsetup-util.c

```diff
@@ -0,0 +1,108 @@
+#include <unistd.h>
+
+#include "cryptsetup-util.h"
+#include "fd-util.h"
+#include "format-util.h"
+#include "memory-util.h"
+#include "path-util.h"
+#include "stat-util.h"
+#include "strv.h"
+
+#define KEY_FILE_SIZE_MAX (16U*1024U*1024U) /* 16 MiB */
+
+int load_key_file(
+                const char *key_file,
+                char **search_path,
```

Comment:
The usual comment for the future: it's much easier to review stuff when the move is done in a separate commit from the additions...